compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -456,6 +456,39 @@ declare_lint! {
     "unrecognized lint attribute"
 }
 
+declare_lint! {
+    /// The `unfulfilled_lint_expectation` lint detects lint trigger expectations
+    /// that have not been fulfilled.
+    ///
+    /// ### Example
+    ///
+    /// ```rust
+    /// #![feature(lint_reasons)]
+    ///
+    /// #[expect(unused_variables)]
+    /// let x = 10;
+    /// println!("{}", x);
+    /// ```
+    ///
+    /// {{produces}}
+    ///
+    /// ### Explanation
+    ///
+    /// It was expected that the marked code would emit a lint. This expectation
+    /// has not been fulfilled.
+    ///
+    /// The `expect` attribute can be removed if this is intended behavior otherwise
+    /// it should be investigated why the expected lint is no longer issued.
+    ///
+    /// Part of RFC 2383. The progress is being tracked in [#54503]
+    ///
+    /// [#54503]: https://github.com/rust-lang/rust/issues/54503
+    pub UNFULFILLED_LINT_EXPECTATION,
```

Comment:
The RFC named this lint `expectation_missing`, though it said the name was open to change. I like `unfulfilled_lint_expectation` but it should use the plural form: `unfulfilled_lint_expectations`. See the [lint naming guidelines](https://rustc-dev-guide.rust-lang.org/diagnostics.html#lint-naming) in the rustc dev guide.

Reply:
This name fits better with the naming guidelines, as `#[allow(unfulfilled_lint_expectations)]` makes more sense than `#[allow(expectation_missing)]`. I missed the plural part, nice catch!